backend/mlarchive/archive/utils.py

```diff
@@ -212,17 +213,58 @@ def get_known_mailman_lists(private=None):
     return mlists
 
 
+def get_mailman_lists(private=None):
+    '''Returns EmailLists that are managed by mailman 3.
+    Specify list.private value or leave out to retrieve all lists.
+    '''
+    url = settings.MAILMAN_API_LISTS
+    try:
+        response = requests.get(
+            url,
+            auth=(settings.MAILMAN_API_USER, settings.MAILMAN_API_PASSWORD),
+            timeout=settings.DEFAULT_REQUESTS_TIMEOUT)
+    except requests.Timeout as exc:
+        logger.error(f'GET request timed out for [{url}]: {exc}')
+        return []
```

Comment:
If I read how this (and get_subscribers) is used correctly, a transient failure will empty your data structures. Raising and handling an exception may be a better thing to do in these cases.